runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/internal/ReconfigureIT.java

```diff
@@ -0,0 +1,139 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.engine.internal;
+
+import static io.aklivity.zilla.runtime.engine.EngineConfiguration.ENGINE_DRAIN_ON_CLOSE;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.rules.RuleChain.outerRule;
+
+import java.io.File;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.DisableOnDebug;
+import org.junit.rules.TestRule;
+import org.junit.rules.Timeout;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+import org.openjdk.jmh.util.FileUtils;
+
+import io.aklivity.zilla.runtime.engine.test.EngineRule;
+import io.aklivity.zilla.runtime.engine.test.annotation.Configuration;
+
+
+public class ReconfigureIT
+{
+    private final K3poRule k3po = new K3poRule()
+        .addScriptRoot("net", "io/aklivity/zilla/specs/engine/streams/network")
+        .addScriptRoot("app", "io/aklivity/zilla/specs/engine/streams/application");
+
+    private final TestRule timeout = new DisableOnDebug(new Timeout(10, SECONDS));
+
+    private final EngineRule engine = new EngineRule()
+        .directory("target/zilla-itests")
+        .commandBufferCapacity(1024)
+        .responseBufferCapacity(1024)
+        .counterValuesBufferCapacity(8192)
+        .configure(ENGINE_DRAIN_ON_CLOSE, false)
+        .configurationRoot("io/aklivity/zilla/runtime/engine/internal")
+        .external("app0")
+        .external("app1")
+        .clean();
+
+    @Rule
+    public final TestRule chain = outerRule(engine).around(k3po).around(timeout);
+
+    @After
+    public void cleanupFileSystem() throws Exception
+    {
+        String source = new File(ReconfigureIT.class
+            .getResource("zilla.reconfigure.original.json")
+            .toURI()).getAbsolutePath();
+        String target = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.json").getAbsolutePath();
+        FileUtils.copy(source, target);
+        new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.missing.json").delete();
+    }
+
+    @Test
+    @Configuration("zilla.reconfigure.json")
+    @Specification({
+        "${app}/client.sent.data.reconfigure.modify/server",
+        "${net}/client.sent.data.reconfigure.modify/client"
+    })
+    public void shouldConnectToZillaOnNewPortWhenConfigModified() throws Exception
+    {
+        k3po.start();
+
+        String source = new File(ReconfigureIT.class
+            .getResource("zilla.reconfigure.after.json")
+            .toURI()).getAbsolutePath();
+        String target = new File("target/test-classes/" +
+            "io/aklivity/zilla/runtime/engine/internal/zilla.reconfigure.json").getAbsolutePath();
+        FileUtils.copy(source, target);
```

Comment:
```
import java.nio.file.Files;
```
Define some useful fields, designed to survive a refactoring.
```
private final String packageName = ReconfigureIT.class.getPackageName();
private Path configDir = Paths.get("target/test-classes", packageName.replace(".", "/"));
```
Use Files.copy to read from classpath and write to filesystem.
```
InputStream source = ReconfigureIT.class.getResourceAsStream("zilla.reconfigure.after.json");
Path target = configDir.resolve("zilla.reconfigure.json");
Files.copy(source, target);
```

Similar changes for the other methods.